clang/lib/Lex/LiteralSupport.cpp

```diff
@@ -1117,6 +1118,31 @@ NumericLiteralParser::NumericLiteralParser(StringRef TokSpelling,
       if (isImaginary) break;   // Cannot be repeated.
       isImaginary = true;
       continue;  // Success.
+    case '_':
+      if (isFPConstant)
+        break; // Invalid for floats
+      if (HasSize)
+        break;
+      if (DoubleUnderscore)
+        break; // Cannot be repeated.
```

Comment:
Hi @js324. our static verifier is reporting this 'break' as dead code saying that it will always be false. I removed all references to DoubleUnderscore and ran the lit tests and there are no fails. Do you have a test case in mind that exercises this condition? If so we should add a test for it. If not we can simplify this code and remove DoubleUnderscore. What do you think?

Reply:
This is presumably because there currently is only one suffix that uses the `DoubleUnderscore` flag, so the other flags that also get set when `__wb` is parsed (e.g. `HasSize`) would be enough to disallow repetition here, from what I can tell. I’d still keep `DoubleUnderscore` in case we decide to add more literal suffixes that contain double underscores.

Reply:
Hmm.. I'd definitely prefer not to have this be forgotten if we add a second double underscore literal suffix, but presumably there is something we could do to make it not active code.  Perhaps we could replace it with a `assert(!DoubleUnderscore && "Doubleunderscore should be handled like above if we ever get a suffix that could hit this")`

WDYT?  I don't have the ability to do so, but just a suggestion and a 'patches welcome' to whoever wants to do the work (@js234 @mikerice1969 @Sirraide ).

Reply:
An assertion would also make sense yeah.

> whoever wants to do the work

(For the record, I’m busy refactoring AST visitors so not me)

Reply:
> I'd definitely prefer not to have this be forgotten if we add a second double underscore literal suffix

And yeah, agreed.

Reply:
Thanks for the comments. Submitted https://github.com/llvm/llvm-project/pull/96579